postgres_exporter/common/pg13/setup.sql

```diff
@@ -407,5 +407,49 @@ CREATE VIEW monitor.pg_hba_hash AS
     FROM hba_ordered_list;
 
 
+
+DROP TABLE IF EXISTS monitor.pg_stat_statements_info;
+-- Table to store last reset time for pg_stat_statements
+CREATE TABLE monitor.pg_stat_statements_info(
+   reset_time timestamptz 
+);
+
+DROP FUNCTION IF EXISTS monitor.ccp_pg_stat_statements_reset(int);
+-- Function to reset pg_stat_statements periodically
+CREATE FUNCTION monitor.ccp_pg_stat_statements_reset(p_throttle_minutes integer DEFAULT 10)
+  RETURNS bigint
+  LANGUAGE plpgsql
+  SECURITY DEFINER
+AS $function$
+DECLARE
+
+  v_reset_timestamp      timestamptz;
+  v_throttle             interval;
+ 
+BEGIN
+
+  IF p_throttle_minutes < 0 THEN
+      RETURN 0;
+  END IF;
+
+  v_throttle := make_interval(mins := p_throttle_minutes);
+
+  SELECT COALESCE(max(reset_time), '1970-01-01'::timestamptz) INTO v_reset_timestamp FROM monitor.pg_stat_statements_info;
+
+  IF ((CURRENT_TIMESTAMP - v_reset_timestamp) > v_throttle) THEN
+      -- Ensure table is empty 
+      DELETE FROM monitor.pg_stat_statements_info;
+      PERFORM pg_stat_statements_reset();
+      INSERT INTO monitor.pg_stat_statements_info(reset_time) values (now());
+  END IF;
+
+  RETURN (SELECT extract(epoch from reset_time) FROM monitor.pg_stat_statements_info);
```

Comment:
May be using the reset time to display how "stale" the metrics are somewhere. I'd have to see it running to double-check. @pmereddy1 can you confirm?

Reply:
Actually, that would always be the time when the reset function was called. So, yeah, this should either just return 1 or zero to indicate whether the stats were reset or not.

Reply:
@keithf4 @sharmay : reset_time metric is not being used by grafana at this point. If successful, this function will return the last stats reset time. Otherwise, it will return a 0.

Reply:
At this point I don't myself. It's not a metric that's being returned for monitoring, just a return when the function is called. If we need it, can easily be added back in.

Reply:
If I change the function to return 1 or 0 to indicate whether the stats were reset or not, the metric name will not be correct. Changing the metric name will cascade to a few other places

Reply:
Sorry, I got this mixed up with what the backrest metric does (it queries the table, not the function). I forgot the pg_stats metric actually calls the reset function itself so it does need to return the time. Can leave this as is.

Reply:
> Do you see no value in returning the actual time when the stats were reset?

There is not benefit in returning actual time of reset. Reset only happens within function and this function is called during metric collection and each metric has time associated.

Reply:
> @keithf4 @sharmay : reset_time metric is not being used by grafana at this point. If successful, this function will return the last stats reset time. Otherwise, it will return a 0.

just return 0 or 1.

Reply:
> > Do you see no value in returning the actual time when the stats were reset?
> 
> There is not benefit in returning actual time of reset. Reset only happens within function and this function is called during metric collection and each metric has time associated.

The metrics are not reset each time the metrics are scraped though. So returning the time since they were actually reset could be useful information to provide or even monitor for to ensure they don't get too stale.